### ISSUE TITLE
Custom Maps in Admin breaks UI if one map URL is not working

### ISSUE THREAD

**Issue by flamber** [metabase/metabase#12599]
**Describe the bug**
In Admin > Settings > Maps, if one of the Custom Maps have an URL that isn't working (domain gone, server down, malformed GeoJSON, etc), then it's not possible to add any new maps, and the modal can break UI, which requires a browser refresh to fix.

**Workaround**
- Hover the failing map and click Remove-button. Do a browser refresh to confirm.
- If that doesn't work, then you need to remove the map from the Metabase application database.
Look for the row `custom-geojson` in table `setting`.

**To Reproduce**
1. In application database `setting` table, change value for `custom-geojson` to:
```
{"5f2e781e-a9b8-c1c3-43bb-3a64628ab37f":{"name":"Failing","url":"http://fail.example.com/geo.json","region_key":"id","region_name":"name"},"7702d6dc-2eb5-b309-14bb-8707e16ca770":{"name":"Working","url":"https://raw.githubusercontent.com/metabase/metabase/master/resources/frontend_client/app/assets/geojson/us-states.json","region_key":"STATE","region_name":"NAME"}}
```
2. Start Metabase (required to do step 1 first, since settings are memorized, so can't hot-change)
3. Go to Admin > Settings > Maps
4. Click "Working" map, which will display the error ```Unable to connect to unknown host at URL `http://fail.example.com/geo.json` ``` (**notice** the URL is from the "Failing" map)
![image](https://user-images.githubusercontent.com/1447303/83055214-d5ec8a00-a053-11ea-8d29-4b2d0e8afb58.png)
5. Click "Cancel" or "Save map" button doesn't work, browser console `uncaught exception: Object` - UI completely broken, browser refresh needed
6. Click "Add a map", input an URL and click "Load", which fails like step 4 and 5
7. Hover "Working" map and click "Remove" button does not work, browser console `uncaught exception: Object`
8. Hover "Failing" map and click "Remove" button - now it's possible to add/remove other maps

<details><summary>stack of error</summary>

```
05-27 20:08:27 ERROR middleware.log :: PUT /api/setting/custom-geojson 500 116.4 ms (0 DB calls) 
{:message "Unable to connect to unknown host at URL `http://fail.example.com/geo.json`",
 :type java.lang.Exception,
 :stacktrace
 ["--> api.geojson$valid_json_url_string_QMARK_.invokeStatic(geojson.clj:65)"
  "api.geojson$valid_json_url_string_QMARK_.invoke(geojson.clj:53)"
  "api.geojson$fn__56258.invokeStatic(geojson.clj:80)"
  "api.geojson$fn__56258.invoke(geojson.clj:79)"
  "api.geojson$validate_custom_geo_json.invokeStatic(geojson.clj:106)"
  "api.geojson$validate_custom_geo_json.invoke(geojson.clj:103)"
  "api.geojson$fn__56277$fn__56278.invoke(geojson.clj:115)"
  "models.setting$set_BANG_.invokeStatic(setting.clj:393)"
  "models.setting$set_BANG_.invoke(setting.clj:378)"
  "api.setting$fn__65013.invokeStatic(setting.clj:34)"
  "api.setting$fn__65013.invoke(setting.clj:28)"
  "middleware.auth$enforce_authentication$fn__66808.invoke(auth.clj:14)"
  "routes$fn__68332$fn__68333.doInvoke(routes.clj:56)"
  "middleware.exceptions$catch_uncaught_exceptions$fn__66907.invoke(exceptions.clj:97)"
  "middleware.exceptions$catch_api_exceptions$fn__66904.invoke(exceptions.clj:85)"
  "middleware.log$log_api_call$fn__68710$fn__68711.invoke(log.clj:197)"
  "middleware.log$log_api_call$fn__68710.invoke(log.clj:191)"
  "middleware.security$add_security_headers$fn__66870.invoke(security.clj:121)"
  "middleware.json$wrap_json_body$fn__68413.invoke(json.clj:64)"
  "middleware.json$wrap_streamed_json_response$fn__68431.invoke(json.clj:100)"
  "middleware.misc$maybe_set_site_url$fn__68740.invoke(misc.clj:59)"
  "middleware.session$bind_current_user$fn__63397$fn__63398.invoke(session.clj:209)"
  "middleware.session$do_with_current_user.invokeStatic(session.clj:184)"
  "middleware.session$do_with_current_user.invoke(session.clj:177)"
  "middleware.session$bind_current_user$fn__63397.invoke(session.clj:208)"
  "middleware.session$wrap_current_user_id$fn__63382.invoke(session.clj:161)"
  "middleware.session$wrap_session_id$fn__63326.invoke(session.clj:125)"
  "middleware.auth$wrap_api_key$fn__66816.invoke(auth.clj:27)"
  "middleware.misc$bind_user_locale$fn__68745.invoke(misc.clj:78)"
  "middleware.misc$add_content_type$fn__68726.invoke(misc.clj:29)"
  "middleware.misc$disable_streaming_buffering$fn__68754.invoke(misc.clj:94)"
  "server$async_proxy_handler$fn__68488.invoke(server.clj:70)"
  "server.proxy$org.eclipse.jetty.server.handler.AbstractHandler$ff19274a.handle(Unknown Source)"]}
```

</details>

**Information about your Metabase Installation:**
Metabase `master`, 0.35.3 thru 0.32.10 (didn't test below that)

**Additional context**
_Originally posted in #12597_

**Comment by Murphysc** [comment]
Just a note on this, in my situation the "Remove" button would not remove the failing map. Despite it seeming like the button worked, upon reload no map were removed, thus still not being able to add/remove other maps.

**Comment by camsaul** [comment]
Fixed by #13565
